control/control.go

```diff
@@ -41,17 +45,35 @@ type HTTPServer struct {
 // NewHTTPServer initializes a new control server for manipulating
 // ContainerPilot's runtime configuration.
 func NewHTTPServer(cfg *Config) (*HTTPServer, error) {
-	if cfg == nil {
-		err := errors.New("control server not loading due to missing config")
-		return nil, err
-	}
 	srv := &HTTPServer{
 		Addr: cfg.SocketPath,
 	}
+	if err := srv.Validate(); err != nil {
+		return nil, err
+	}
+
 	srv.InitRx()
 	return srv, nil
 }
 
+// Validate validates the state of the control server and ensures that the
+// socket does not exist prior to setting up the listener (bind).
+func (srv HTTPServer) Validate() error {
+	if srv.Addr == "" {
+		return MissingAddrErr
+	}
+
+	// TODO(justinwr@): maybe we should check file ownership and/or perms
+	if _, err := os.Stat(srv.Addr); err == nil {
+		log.Debugf("control: unlinking previous socket at %s", srv.Addr)
+		if err := os.Remove(srv.Addr); err != nil && os.IsExist(err) {
```

Comment:
We've got `os.IsExist(err)` and my hope was that the file would be found and removed under most normal conditions. Do you think if there's any error we should return it, removing the call to `IsExist`?